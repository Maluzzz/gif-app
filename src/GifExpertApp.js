import React, { useState } from 'react'
import { CategorieAdd } from './components/CategorieAdd'
import { GifResults } from './components/GifResults'

export const GifExpertApp = ({ defaultCategories=[] }) => {
  const [categories, setCategories] = useState(defaultCategories)

  return (
    <div className='main'>
      <h2> Gifs App <span>🧐</span></h2>
      <CategorieAdd setCategories={ setCategories }></CategorieAdd>
      {categories.map((category) => (
        <GifResults category={ category } key={ category } />
      ))}
      <div className='footer'><a href='https://github.com/Maluzzz/'>Maluzzz ♥</a></div>
    </div>
  )
}
