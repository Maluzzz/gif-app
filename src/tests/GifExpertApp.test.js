import '@testing-library/jest-dom'

import { shallow } from 'enzyme'

import { GifExpertApp } from '../GifExpertApp'

describe('testing GifExpertApp', () => {

      test('should render the component', () => {

            const wrapper = shallow(<GifExpertApp />)
            expect(wrapper).toMatchSnapshot()
      })

      test('should render list of categories', () => {

            const categories = ['Marvel', 'IronMan']
            const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
            expect(wrapper).toMatchSnapshot()
            expect(wrapper.find('GifResults').length).toBe(categories.length)
      })
})
