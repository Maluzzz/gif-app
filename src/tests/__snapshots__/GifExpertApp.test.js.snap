// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GifExpertApp should render list of categories 1`] = `
<div
  className="main"
>
  <h2>
    Search Gifs App 
    <span>
      üßê
    </span>
  </h2>
  <CategorieAdd
    setCategories={[Function]}
  />
  <GifResults
    category="Marvel"
    key="Marvel"
  />
  <GifResults
    category="IronMan"
    key="IronMan"
  />
  <div
    className="footer"
  >
    <a
      href="https://github.com/Maluzzz/"
    >
      Maluzzz ‚ô•
    </a>
  </div>
</div>
`;

exports[`testing GifExpertApp should render the component 1`] = `
<div
  className="main"
>
  <h2>
    Search Gifs App 
    <span>
      üßê
    </span>
  </h2>
  <CategorieAdd
    setCategories={[Function]}
  />
  <div
    className="footer"
  >
    <a
      href="https://github.com/Maluzzz/"
    >
      Maluzzz ‚ô•
    </a>
  </div>
</div>
`;
